objects/src/block/header.rs

```diff
@@ -275,17 +272,111 @@ impl Deserializable for BlockHeader {
     }
 }
 
-// UTILITIES
+// BLOCK NUMBER
 // ================================================================================================
 
-/// Returns the block number of the epoch block for the given `epoch`.
-pub const fn block_num_from_epoch(epoch: u16) -> u32 {
-    (epoch as u32) << BlockHeader::EPOCH_LENGTH_EXPONENT
+/// A convenience wrapper around a `u32` representing the number of a block.
+///
+/// Each block has a unique number and block numbers increase monotonically by `1`.
+#[derive(Debug, Default, Eq, PartialEq, Copy, Clone, PartialOrd, Ord, Hash)]
+pub struct BlockNumber(u32);
```

Comment:
This type and its method have grown a bit since the start of this PR. I think I would move this out into a separate module, i.e. `objects/src/block/block_number.rs` and add 

```rust
mod block_number;
pub use block_number::BlockNumber
```

in `objects/src/block/mod.rs`